File: Front-end/health-mind-system-frontend/src/shared/types/dtos/PlanoSaude.dto.ts
import { StatusPlanoSaudeEnum } from "../../domain/enums/status-plano-saude.enum";
import { CoberturaPlanoDTO } from "./CoberturaPlano.dto";

export interface PlanoSaudeDTO {
    id?: string,
    nome: string,
    codigo: string,
    statusPlanoSaude: StatusPlanoSaudeEnum,
    telefone: string,
    email: string,
    coberturasPlanoDTO?: CoberturaPlanoDTO[]
}

File: Back-end/HealthMindBackend/HealthMindBackend.Application/DTOs/PlanoSaudeDTO.cs
using HealthMindBackend.Domain.Enums;
using HealthMindBackend.Domain.ValueObjects.Contato;
using HealthMindBackend.Domain.ValueObjects.Financeiro.CoberturaPlano;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HealthMindBackend.Application.DTOs
{
    public class PlanoSaudeDTO
    {
        public String? Id { get; set; }
        public String Nome { get; set; }
        public String Codigo { get; set; }
        public StatusPlanoSaudeEnum StatusPlanoSaude { get; set; }
        public String Telefone { get; set; }
        public String Email { get; set; }
        public ICollection<CoberturaPlanoDTO>? CoberturasPlanoDTO { get; set; }
    }
}
